scripts/check_integrity.py

```diff
@@ -0,0 +1,251 @@
+#! /bin/python
+import bibtexparser
+from bibtexparser.bparser import BibTexParser
+import re
+import yaml
+import pandas as pd
+from titlecase import titlecase
+import os
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+database_file = '../literature.bib'
+
+title_regex = re.compile(r"^\{.*\}$", re.DOTALL)
+pages_regex = re.compile(r"^[a-zA-Z0-9,.:()]+(--[a-zA-Z0-9,.:()]+)?$", re.DOTALL)
+author_regex = re.compile(r"([^,]+),[^,]+(\s+and\s+([^,]+),[^,]+)*", re.DOTALL)
+
+
+def get_path(file_name):
+    return os.path.join(script_dir, file_name)
+
+
+## Checks if parameter is set in config file
+def is_set(key):
+    if not key in settings:
+        print('WARNING settings ' + key + ' is not defined!')
+        return False
+    return settings[key]
+
+
+## Adds a problem to the list of problems of a given entry
+def add_problem(entry, problem_type, problem_message):
+    entry_id = entry['ID']
+    if entry_id not in problems:
+        problems[entry_id] = []
+    problems[entry_id].append((problem_type, problem_message))
+
+
+## Checks if fields declared as string fields contain string values
```

Comment:
This comment was very confusing at first. field = entry? Can you leave a small example here?